src/coreclr/vm/stringliteralmap.cpp

```diff
@@ -485,7 +501,7 @@ StringLiteralEntry *GlobalStringLiteralMap::AddStringLiteral(EEStringData *pStri
     {
     PinnedHeapHandleBlockHolder pStrObj(&m_PinnedHeapHandleTable,1);
```

Comment:
@Maoni0 Do we need the pinned heap handle for the frozen strings? Would it be ok to not allocate the pinned heap handle for the frozen strings so that the GC has less work to do?

Reply:
For frozen strings these handles are used to basically avoid duplications (hashmap) in the frozen segments. But I assume you mean that they can be allocated outside of GC heap with normal alloc

Reply:
> I assume you mean that they can be allocated outside of GC heap with normal alloc

Yes, it is what I meant.

Reply:
It can be StringLiteralEntry. 

We can have `STRINGREF*                  m_pStringObj;` for heap allocated string and `STRINGREF                  m_pStringObj;` for pinned strings.